python/cudf/utils/utils.py

```diff
@@ -178,3 +179,24 @@ def buffers_from_pyarrow(pa_arr, dtype=None):
             np.empty(0, dtype=new_dtype)
         )
     return (pamask, padata)
+
+
+def cudf_dtype_from_pydata_dtype(dtype):
+    """ Given a numpy or pandas dtype, converts it into the equivalent cuDF
+        Python dtype.
+    """
+    try:
+        # pd 0.24.X
+        from pandas.core.dtypes.common import \
+            infer_dtype_from_object
+    except ImportError:
+        # pd 0.23.X
+        from pandas.core.dtypes.common import \
+            _get_dtype_from_object as infer_dtype_from_object
+
+    if pd.api.types.is_categorical_dtype(dtype):
+        pass
```

Comment:
I don't follow why you need an if statement with a pass inside of it.

Reply:
If dtype is a categorical, the `np.issubdtype` will throw an exception instead of return `False` since it doesn't know how to interpret it.